### PR TITLE
Update FIAM for iOS 13 Scene lifecycle paradigm

### PR DESCRIPTION
**Pull request by christibbs**
Two key fixes:
- Throw both `UIScene` and `UIApplication` lifecycle notifications
- Update `UIWindow` creation logic to initialize based on a foregrounded scene (if iOS 13 is available)

Fixes https://github.com/firebase/firebase-ios-sdk/issues/3524 with the exception around concerns with opening URL from a scene. From testing, though, current URL opening logic still works as intended.

### CODE REVIEW

File: Firebase/InAppMessaging/Flows/FIRIAMActivityLogger.m [hunk 1]
```diff
@@ -106,7 +106,14 @@ - (instancetype)initWithMaxCountBeforeReduce:(NSInteger)maxBeforeReduce
                                              selector:@selector(appWillBecomeInactive)
                                                  name:UIApplicationWillResignActiveNotification
                                                object:nil];
-
+#if defined(__IPHONE_13_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+    if (@available(iOS 13.0, *)) {
+      [[NSNotificationCenter defaultCenter] addObserver:self
```
Comment (ryanwilson) [inline Firebase/InAppMessaging/Flows/FIRIAMActivityLogger.m]: Are there situations where `appWillBecomeInactive` could be called twice, i.e. could `UIApplicationWillResignActiveNotification` and `UISceneWillDeactivateNotification` both be fired for the same app close?

File: Firebase/InAppMessagingDisplay/FIDRenderingWindowHelper.m [hunk 2]
```diff
@@ -24,7 +24,16 @@ + (UIWindow *)UIWindowForModalView {
   static dispatch_once_t onceToken;
 
   dispatch_once(&onceToken, ^{
-    UIWindowForModal = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
+#if defined(__IPHONE_13_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+    if (@available(iOS 13.0, *)) {
+      UIWindowScene *foregroundedScene = [[self class] foregroundedScene];
```
Comment (ryanwilson) [inline Firebase/InAppMessagingDisplay/FIDRenderingWindowHelper.m]: Could this ever be `nil` on iOS 13?

Comment (christibbs) [reply Firebase/InAppMessagingDisplay/FIDRenderingWindowHelper.m]: Yes, could be `nil` if this is called there are no foregrounded scenes. In that case it should be a no-op. Though, now that you mention it I should force the nil case to make sure nothing weird happens.

File: Firebase/InAppMessaging/Analytics/FIRIAMClearcutLogStorage.m [hunk 3]
```diff
@@ -92,6 +92,15 @@ - (instancetype)initWithExpireAfterInSeconds:(NSInteger)expireInSeconds
                                              selector:@selector(appWillBecomeInactive)
                                                  name:UIApplicationWillResignActiveNotification
                                                object:nil];
+#if defined(__IPHONE_13_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+    if (@available(iOS 13.0, *)) {
+      [[NSNotificationCenter defaultCenter] addObserver:self
+                                               selector:@selector(appWillBecomeInactive)
```
Comment (mikehaney24) [inline Firebase/InAppMessaging/Analytics/FIRIAMClearcutLogStorage.m]: ideally, nsnotification observers have a single argument, the notification.

File: Firebase/InAppMessaging/Flows/FIRIAMActivityLogger.m (hunk 1 above)
Comment (mikehaney24) [reply Firebase/InAppMessaging/Flows/FIRIAMActivityLogger.m]: Same thing for becoming active.

File: Firebase/InAppMessagingDisplay/FIDRenderingWindowHelper.m (hunk 2 above)
Comment (christibbs) [reply Firebase/InAppMessagingDisplay/FIDRenderingWindowHelper.m]: Tested this, it's a no-op as expected.

**Pull request merged**